.spin/cmds.py

```diff
@@ -263,6 +263,52 @@ def _run_asv(cmd):
 
     util.run(cmd, cwd='benchmarks', env=env)
 
+@click.command()
+@click.option(
+    "-b", "--branch",
+    metavar='branch',
+    default="main",
+)
+@click.option(
+    '--uncommitted',
+    is_flag=True,
+    default=False,
+    required=False,
+)
+@click.pass_context
+def lint(ctx, branch, uncommitted):
+    """🔦 Run lint checks on diffs.
+    Provide target branch name or `uncommitted` to check changes before committing:
+
+    \b
+    Examples:
+
+    \b
+    For lint checks of your development brach with `main` or a custom branch:
+
+    \b
+    $ spin lint # defaults to main
+    $ spin lint --branch custom_branch
+
+    \b
+    To check just the uncommitted changes before committing
+
+    \b
+    $ spin lint --uncommitted
+    """
+    click.secho(
+        "Performing lint checks " + "for uncommitted changes" if uncommitted
+        else f"against {branch} branch",
+        bold=True, fg="bright_green",
+    )
+    try:
+        from tools.linter import DiffLinter
+    except ModuleNotFoundError as e:
+        raise click.ClickException(
+            f"{e.msg}. Install using linter_requirements.txt"
+        )
```

Comment:
This doesn't look correct. I see this, while `python tools/linter.py` works fine:
```
$ spin lint
against main branch
Error: No module named 'tools'. Install using linter_requirements.txt
```

You cannot use absolute imports like that to something that isn't a package. Instead, you want to use `importlib` functionality here. Search for `importlib.util.spec_from_file_location`, there's a bunch of places in the code base where we use that.

Reply:
I've fixed it in #24986, which would probably lead to a conflict. I'll resolve the conflict on whichever goes second